docs/operating-scylla/procedures/cluster-management/add-node-to-cluster.rst

```diff
@@ -98,11 +98,18 @@ Procedure
         UN  192.168.1.202  91.11 KB   256     32.9%             125ed9f4-7777-1dbn-mac8-43fddce9123e   B1
         UN  192.168.1.203  124.42 KB  256     32.6%             675ed9f4-6564-6dbd-can8-43fddce952gy   B1
 
-#. When the new node status is Up Normal (UN), use :doc:`nodetool cleanup </operating-scylla/nodetool-commands/cleanup>` cleanup command on all the nodes in the cluster except the new node that has just been added. It will remove keys that no longer belong to the node. Run this command one node at a time. It is possible to postpone this step to low demand hours.
+#. When the new node status is Up Normal (UN), run the :doc:`nodetool cleanup </operating-scylla/nodetool-commands/cleanup>` command on all nodes in the cluster except for the new node that has just been added. Cleanup removes keys that were streamed to the newly added node and are no longer owned by the node.
 
     .. note::
 
-       If you are using Scylla Enterprise 2018.1.5 or earlier or Scylla Open source 2.3 or earlier, **do not** run the ``nodetool cleanup`` command before upgrading to the latest release of your branch. See this `issue <https://github.com/scylladb/scylla/issues/3872>`_ for more information.
+       To prevent data resurrection, it's essential to complete cleanup after adding nodes and before any node is decommissioned or removed.
+       However, cleanup may consume significant resources. Use the following guideline to reduce cleanup impact:
+
+       Tip 1: When adding multiple nodes, run the cleanup operations after all nodes are added on all nodes but the last one to be added.
+
+       Tip 2: Postpone cleanup to low demand hours while ensuring it completes successfully before any node is decommissioned or removed.
+
+       Tip 3: Run cleanup one node at a time, reducing overall cluster impact.
```

Comment:
@bhalevy Tips above are all good and are absolutely correct in a general case.
I'm personally preaching for exactly same approach (except for Tip 3) in our automation - KISS principle! ;)

However I wonder if it makes sense to provide a few more "exceptions tips" to this doc?
For example:
1) In the context of a particular table and its specific configuration you don't have to run cleanup after adding node(s) an before decommissioning IF the time that passes between the beginning of the first node addition and until the completion of decommissioning of the last node is shorter than `gc_grace_seconds`.
2) The same holds if there are not DELETEs at all, e.g. all data uses TTL.

Reply:
@vladzcloudius IMHO its better not to list these edge cases:
- Its easy to get them wrong 
- the risk of running  a redundant cleanup is low

Reply:
Makes sense.

Reply:
We're actually switching SCT to run them in parallel, since it's waste a lot testing time when doing them one by one.

Reply:
> We're actually switching SCT to run them in parallel, since it's waste a lot testing time when doing them one by one.

@roydahan it's not like you are forbidden do it parallel - it's just that it's going to make those nodes work harder (it's similar to a compaction running on all shards at the same time).

The "tip" is just a tip - you can use it or not. It's up to you.

Reply:
> @vladzcloudius IMHO its better not to list these edge cases:
> 
>     * Its easy to get them wrong
> 
>     * the risk of running  a redundant cleanup is low

+1

Reply:
> > In addition to the tips, I created two PRs for ideas to prevent data resurrection automatically and trigger cleanup automatically.
> > #12220 #12220
> 
> Which is the second one, @asias ?

It is @12221. I updated above.

Reply:
As a rule, we shouldn't document exceptions and edge cases in the core documentation. They belong to troubleshooting or similar sections.